kale/workflow_widgets.py

```diff
@@ -571,21 +570,19 @@ def _call_read_log(self, caller=None):
     def _run_wrapper(self, *args):
         with self._widget_log:
             try:
-                if not self.future.running():
-                    self.future = self._thread_pool.submit(self.run_workflow)
-                    print("Workflow submitted.")
-                else:
-                    print("Workflow already running.")
+                self.futures.append(self._thread_pool.submit(self.run_workflow))
+                print("Workflow submitted.")
             except AttributeError as e:
                 # Attribute error if self.future is None
                 # which means workflow has not been submitted.
-                self.future = self._thread_pool.submit(self.run_workflow)
+                self.future = [self._thread_pool.submit(self.run_workflow)]
```

Comment:
self.futures vs self.future? looks like this could be a typo

Reply:
Good catch!